P/PETSc/build_tarballs.jl

```diff
@@ -96,56 +109,86 @@ build_petsc()
 
     # See if we can install MUMPS
     USE_MUMPS=0    
-    if [ -f "${libdir}/libdmumps.${dlext}" ] && [ "${1}" == "double" ]; then
+    if [ -f "${libdir}/libdmumps.${dlext}" ] && [ "${1}" == "double" ] && [ "${2}" == "real" ]; then
         USE_MUMPS=1    
         MUMPS_LIB="--with-mumps-lib=${libdir}/libdmumps.${dlext} --with-scalapack-lib=${libdir}/libscalapack32.${dlext}"
         MUMPS_INCLUDE="--with-mumps-include=${includedir} --with-scalapack-include=${includedir}"
     else
         MUMPS_LIB=""
         MUMPS_INCLUDE=""
     fi
+    
+    LIBFLAGS="-L${libdir}" 
+    if [[ "${target}" == *-mingw* ]]; then
+        LIBFLAGS="-L${libdir} -lssp" 
+    fi
+
+    if [[ "${target}" == *-apple* ]]; then 
+        # on a mac, there might otherwise be confusion with clang
+        CC=gcc
+        FC=gfortran
+        CXX=g++
```

Comment:
Not clear to me why we can't use clang, which _is_ what we typically want to use.

Reply:
Sounds like the commit [demonstrate fails when clang is used on apple](https://github.com/JuliaPackaging/Yggdrasil/pull/6533/commits/4d4b46657567d3490d85c344c98f87c3c63bcf55) failed to demonstrate the failure :slightly_smiling_face:

Reply:
ok, it did work